client/src/components/Link/Link.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useInRouterContext, Link as RouterLink } from 'react-router-dom';
+
+/**
+ * This component can be used whenever a component which needs a relative link
+ * might be rendered in both context within a router (e.g. asset admin) and
+ * contexts without a router (e.g. inside an image upload modal).
+ */
+function Link({ children, href, ...props }) {
```

Comment:
This is necessary because the update of react-router resulted in the way routes are added being changed - where the route and url both used to be e.g. `/admin/assets/show/1/edit`, the route is now _relative to the base route_, e.g. `assets/show/1/edit`. This means we have to use a react `Link` component when in a react router context so that the correct full path is on the rendered `<a>` tag's `href` attribute..